src/test/java/ReceiptAnalysisTest.java

```diff
@@ -0,0 +1,111 @@
+import com.google.sps.data.AnalysisResults;
+import com.google.sps.servlets.ReceiptAnalysis;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({AnnotateImageResponse.class, AnnotateImageRequest.class,
+    BatchAnnotateImagesResponse.class, ByteString.class, EntityAnnotation.class, Feature.class,
+    Image.class, ImageAnnotatorClient.class, ReceiptAnalysis.class, URL.class})
+public final class ReceiptAnalysisTest {
+  private static final ByteString IMAGE_BYTES = ByteString.copyFromUtf8("byte string");
+  private static final String RAW_TEXT = "raw text";
+
+  @Mock private AnnotateImageRequest request;
+  @Mock private Feature feature;
+  @Mock private Image image;
+
+  @Mock(answer = Answers.RETURNS_SELF) private AnnotateImageRequest.Builder requestBuilder;
+  @Mock(answer = Answers.RETURNS_SELF) private Feature.Builder featureBuilder;
+  @Mock(answer = Answers.RETURNS_SELF) private Image.Builder imageBuilder;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void serveImageTextUrlReturnsAnalysisResults() throws IOException {
+    URL url = mock(URL.class);
+    InputStream inputStream = new ByteArrayInputStream(IMAGE_BYTES.toByteArray());
+    when(url.openStream()).thenReturn(inputStream);
+
+    mockStatic(ByteString.class);
+    when(ByteString.readFrom(inputStream)).thenReturn(IMAGE_BYTES);
+
+    when(imageBuilder.build()).thenReturn(image);
+    mockStatic(Image.class);
+    when(Image.newBuilder()).thenReturn(imageBuilder);
+
+    when(featureBuilder.build()).thenReturn(feature);
+    mockStatic(Feature.class);
+    when(Feature.newBuilder()).thenReturn(featureBuilder);
+
+    when(requestBuilder.build()).thenReturn(request);
+    mockStatic(AnnotateImageRequest.class);
+    when(AnnotateImageRequest.newBuilder()).thenReturn(requestBuilder);
+
+    ImageAnnotatorClient client = mock(ImageAnnotatorClient.class);
+    mockStatic(ImageAnnotatorClient.class);
+    when(ImageAnnotatorClient.create()).thenReturn(client);
+
+    BatchAnnotateImagesResponse batchResponse = mock(BatchAnnotateImagesResponse.class);
+    when(client.batchAnnotateImages(Mockito.<AnnotateImageRequest>anyList()))
+        .thenReturn(batchResponse);
+
+    AnnotateImageResponse response = mock(AnnotateImageResponse.class);
+    ImmutableList<AnnotateImageResponse> responses = ImmutableList.of(response);
+    when(batchResponse.getResponsesList()).thenReturn(responses);
+
+    EntityAnnotation annotation = mock(EntityAnnotation.class);
```

Comment:
Based on tott/289, I don't think classes like EntityAnnotation or AnnotateImageResponse need to be mocked since they don't have network interaction or file IO.

https://drive.google.com/file/d/0B-yWrS8S9kyqTmFFa1NWVWs5c2c/preview